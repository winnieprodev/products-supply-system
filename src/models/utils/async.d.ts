export interface Error {
  message: string;
}

export interface Async<T> {
  isLoading: boolean;
  error: Error | undefined;
  data: T | undefined;
}
