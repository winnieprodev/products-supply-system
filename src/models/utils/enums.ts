export enum WrapperMode {
  full = "full",
  singleLine = "singleLine",
  tiny = "tiny",
}
