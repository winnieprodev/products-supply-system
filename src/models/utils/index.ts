export * from "./async.d";
export * from "./enums";
