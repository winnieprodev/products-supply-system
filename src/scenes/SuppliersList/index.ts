export { default as SuppliersList } from "./SuppliersList";
