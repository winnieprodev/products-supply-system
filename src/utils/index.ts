export * from "./routeUtils";
export * from "./supplierUtils";
export * from "./testUtils";
