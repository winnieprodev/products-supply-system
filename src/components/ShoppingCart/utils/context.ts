import { Async, Cart, ProductCartInfo } from "models";
import React from "react";
import { mockAsync, mockCart } from "utils";

interface ShoppingCartContextProps {
  cartId: Async<number>;
  cart: Async<Cart> | undefined;
  setCart: (cart: Async<Cart> | undefined) => void;
  setProduct: (product: ProductCartInfo) => void;
  removeProduct: (productId: string) => void;
}

export const EMPTY_CART = { products: [] };
const mockFunction = () => {};

export const ShoppingCartContext = React.createContext<ShoppingCartContextProps>(
  {
    cartId: mockAsync(0),
    cart: mockAsync(mockCart(0)),
    setCart: mockFunction,
    setProduct: mockFunction,
    removeProduct: mockFunction,
  },
);
