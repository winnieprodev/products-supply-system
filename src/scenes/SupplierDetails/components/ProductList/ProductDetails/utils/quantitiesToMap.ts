import { Map } from "immutable";
import { CartQuantity, QuantityType } from "models";

export const quantitiesToMap = (quantities: CartQuantity[]) => {
  return Map<QuantityType, number>(
    quantities.map(({ type, amount }) => [type, amount] as const) as Iterable<
      [QuantityType, number]
    >,
  );
};
