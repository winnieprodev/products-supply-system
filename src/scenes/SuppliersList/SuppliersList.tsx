import {
  AsyncWrapper,
  Breadcrumb,
  EmptyState,
  Grid,
  Searchbar,
} from "components";
import { useGetSuppliersList } from "queries";
import { useMemo, useState } from "react";
import { Link, useHistory, useParams } from "react-router-dom";
import ROUTES, { SuppliersListParams } from "routes";
import { getSupplierTypeName } from "utils";
import { Empty } from "utils/types";

import { SupplierTile } from "./components";

const SuppliersList: React.FC<Empty> = () => {
  const { outletId, supplierType } = useParams<SuppliersListParams>();
  const history = useHistory();

  const [searchValue, setSearchValue] = useState("");

  const suppliersAsync = useGetSuppliersList(outletId, supplierType);

  const selectSupplier = (supplierId: string) =>
    history.push(
      ROUTES.SUPPLIER_DETAILS.use({ supplierId, supplierType, outletId }),
    );

  const filteredData = useMemo(
    () =>
      suppliersAsync.data?.filter(
        ({ title }) =>
          searchValue === "" ||
          title.toLowerCase().includes(searchValue.toLowerCase()),
      ) || [],
    [suppliersAsync, searchValue],
  );
  return (
    <>
      <Searchbar value={searchValue} onChange={setSearchValue} />
      <Breadcrumb>
        <Link to={ROUTES.NEW_ORDER.use({ outletId })}>
          <Breadcrumb.Item>{getSupplierTypeName(supplierType)}</Breadcrumb.Item>
        </Link>
        Select Supplier
      </Breadcrumb>
      <AsyncWrapper
        item={suppliersAsync}
        loadingMessage="Fetching available suppliers"
        errorMessage="Could not fetch available suppliers"
      >
        <EmptyState items={filteredData} searchPhrase={searchValue}>
          <Grid itemsInRow={5} centered={filteredData.length < 5}>
            {filteredData.map((supplier) => (
              <SupplierTile
                key={supplier.id}
                item={supplier}
                onClick={() => selectSupplier(supplier.id)}
              />
            ))}
          </Grid>
        </EmptyState>
      </AsyncWrapper>
    </>
  );
};

export default SuppliersList;
