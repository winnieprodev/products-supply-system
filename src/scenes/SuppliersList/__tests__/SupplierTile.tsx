import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { mockAsync, mockSupplier } from "utils";

import SuppliersList from "../SuppliersList";

// import { useGetSuppliers } from "queries";
jest.mock("queries", () => ({
  useGetSuppliersList: () =>
    mockAsync([
      mockSupplier("test supplier A"),
      mockSupplier("test supplier B"),
    ]),
}));

describe("SuppliersList", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <BrowserRouter>
          <SuppliersList />
        </BrowserRouter>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
