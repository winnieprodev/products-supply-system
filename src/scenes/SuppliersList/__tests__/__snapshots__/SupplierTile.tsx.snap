// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuppliersList Snapshot tests it matches the snapshot 1`] = `
.c7 {
  display: contents;
}

.c7 > .animated {
  opacity: 0;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c3 {
  margin-top: 20px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #F25C05;
  font-size: 15px;
  font-weight: 600;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  font-size: 15px;
  padding-left: 5px;
  padding-right: 5px;
  color: #F25C05;
}

.c6 {
  margin: 30px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 > * > * {
  width: calc(20% - 24px);
  margin: 12px;
}

.c10 {
  background: #F3F3F3;
  border-radius: 10px;
  position: relative;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
}

.c8:hover .c9 {
  background: #EEEEEE;
}

.c11 {
  color: rgba(0,0,0,0.25);
  text-shadow: 0 5px 10px white;
  font-size: 6vw;
  font-weight: bold;
  text-align: center;
  line-height: calc(8vw + 80px);
}

.c12 {
  margin: 8px;
  text-align: center;
  font-size: 16px;
  font-weight: bold;
}

.c0 {
  border-radius: 10px;
  height: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #EEEEEE;
  padding: 6px 12px;
  position: relative;
  -webkit-transition: background ease 0.2s;
  transition: background ease 0.2s;
}

.c0:hover,
.c0:active {
  background: #F3F3F3;
}

.c1 {
  font-size: 22px;
  padding-right: 12px;
  border-right: 1px solid #F25C05;
  margin-right: 12px;
}

.c2 {
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  padding-left: 60px;
  width: calc(100% - 60px);
  outline: none;
  border: none;
  background: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
    </div>
    <input
      class="c2"
      placeholder="Search for a supplier or a product"
      value=""
    />
  </div>
  <div
    class="c3"
  >
    <a
      href="/browse/undefined/"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-right fa-w-8 "
            data-icon="angle-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 256 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </a>
    Select Supplier
  </div>
  <div
    class="c6"
  >
    <div
      class="c7"
    >
      <div
        class="c8 animated"
      >
        <div
          class="c9 c10"
        >
          <div
            class="c11"
          >
            tsA
          </div>
        </div>
        <div
          class="c12"
        >
          test supplier A
        </div>
      </div>
      <div
        class="c8 animated"
      >
        <div
          class="c9 c10"
        >
          <div
            class="c11"
          >
            tsB
          </div>
        </div>
        <div
          class="c12"
        >
          test supplier B
        </div>
      </div>
    </div>
  </div>
</div>
`;
